src/stdune/fmt.mli

```diff
@@ -2,6 +2,9 @@ type 'a t = Format.formatter -> 'a -> unit
 
 val list : ?pp_sep:unit t -> 'a t -> 'a list t
 
+(** Like [list], but the printer gets an extra argument for the last element *)
+val list_special : pp_sep:unit t -> (last:bool -> 'a t) -> 'a list t
```

Comment:
I suggest to pass a value of a type `First | Middle | Last` instead. It's often the case that the first element is printed differently as well.

Reply:
It's a bit difficult to define an API for this. What should be passed to the callback when the list is a singleton? If you have existing call sites in mind I can try to generalize a bit based on them.

Reply:
It might be useful to look at ocamlformat's [Fmt.list_fl](https://github.com/ocaml-ppx/ocamlformat/blob/master/src/Fmt.ml#L56) (for first-last) and [Fmt.list_pn](https://github.com/ocaml-ppx/ocamlformat/blob/master/src/Fmt.ml#L42) (for previous-next, to use when it is necessary to peek at the adjacent elements) for reference.

Reply:
I ended up removing that function and do the recursion by hand. there's probably a pattern to extract, but for now this seems pretty specific.